### PR TITLE
Time Considerations

### PR DESCRIPTION
**Pull request by mattjohnsonpint**
Adding time considerations document.

Connects to #73

### CODE REVIEW

**Comment by azurecla** [comment]
Hi __@mj1856__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <p>
        It looks like you're working at Microsoft (mattjoh). If you're full-time, we DON'T require a contribution license agreement.
    </p>
    <p>
        If you are a vendor, or work for Microsoft Open Technologies, DO please sign the electronic contribution license agreement. It will take 2 minutes and there's no faxing! https://cla.azure.com.
    </p>

TTYL, AZPRBOT;

**Comment by mj1856** [comment]
@MikeWasson - Ready for review.  Thanks!

**Comment by MikeWasson** [comment]
Looks great. I made some edits but I like the overall structure. It would be nice to have some more examples of specific IoT scenarios and how one would reason about clocks and timestamps for each. (Like you do for the Fabrikam scenario.) Not sure this doc is the best place for that, though.

**Comment by bennage** [comment]
I think this is good. Any reason we shouldn't merge?

File: docs/articles/time-considerations.md [hunk 1]
```diff
@@ -0,0 +1,394 @@
...
+Windowing functions provide a way to group events that occur within a
+period of time.
+
+If you are using an event stream processor such as Azure Stream Analytics, you should
+be familiar with the different types of windowing functions that are available.
+
+For example, you could use a 1-hour *tumbling* window to see which hours have the most
+activity.  Or, you might use a 5-minute *sliding* window to watch for trends in
+the incoming data across groups of devices.  You can read more about
+windowing functions in the [ASA documentation][asa-windowing].
```
Comment (bennage) [inline docs/articles/time-considerations.md]: Let's replaced ASA with Stream Analytics or Azure Stream Analytics.

**Comment by bennage** [comment]
I do have one minor concern, but we could address it after merging.
We have some .NET specific content. That is fine. However, I would like to structure it differently. Can we group all of the .NET specific content?

**Comment by bennage** [comment]
Ok, one more thought. I'd like to replace the Fabrikam references in the docs with something a bit more generic. My goal is for someone to be able to read this doc indepedently from the rest of the IoTJourney content.

**Comment by bennage** [comment]
After chatting with @MikeWasson, I'd like to adjust the final section a bit. Let's move the Stream Analytics implementation details into the journal entry for long term storage (#192). Likewise, we should broaden the statement here to be more like "different technologies make different assumption about time, windowing, etc. be sure to read their docs..."

**Comment by mj1856** [comment]
:+1: on the changes

**Pull request merged**